### PR TITLE
Feature: Bubble Sort Algorithm 

### PR DESCRIPTION
**Pull request by shadrxcc**
Bubble sort is the sorting of a group of items in an array, where each element is compared to the one next to it and they are swapped if the one on the left is larger than the one on the right. This continues until the array is eventually sorted. This algorithm contains two methods; #Bubble_sort and #Bubble_sort_by methods respectively.
 
- The #Bubble_sort method takes an array and returns a sorted array using the bubble sort methodology. 
- The #Bubble_sort_by method sorts an array by accepting a block with the use of #yield inside the method definition to accomplish this. The block has two arguments that represent the two elements of the array that are being compared.

### CODE REVIEW

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+        fig[a], fig[a + 1] = fig[a + 1], fig[a]
+        sorted = true
+      end
+    end
+    break if sorted
+  end
+  fig
+end
+val = [4, 3, 78, 2, 0, 2]
+print bubble_sort val
```
Comment (soieducation) [inline bubble_sort.rb]: - This method returns [3, 4, 2, 0, 2, 78] instead of [0, 2, 2, 3, 4, 78] when [4,3,78,2,0,2] is passed.

**Review by soieducation — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED 🚦 

## Hello @shadrxcc  :handshake: :handshake: :handshake: 
 Great work so far
- Your bubble_sort_by method sorts the [hi hello hey] array perfectly

### However, there are still changes to be made

# [Required Changes] :warning: 
- [ ] Your bubble_sort method does not return a sorted array, kindly fix it.
- [ ] Your bubble_sort_by method does not return a sorted array for this array [hello hiya hey hi h], kindly fix it.
___
**If you are stuck at any point or have any questions, let us know by leaving a comment here. We will be happy to help**
HAPPY CODING :computer:

File: bubble_sort.rb (hunk 1 above)
Comment (shadrxcc) [reply bubble_sort.rb]: Thank you for the review. Will make necessary changes.

**Review by soieducation — APPROVED** [review]
## Status: Approved :green_circle: 

### Hello @shadrxcc   :wave: 

### You have done an amazing job on this project, congratulations! Your project is approved :clap: :clap: :clap: 

Good luck with your next project :rocket: 

### Remember to submit your project completion form
___
**If you are stuck at any point or have any questions, let us know by leaving a comment here. We will be happy to help**
HAPPY CODING :computer:

**Pull request merged**